### PR TITLE
DM-23484: qserv column headers in results are wrong with optimization turned on

### PR DESCRIPTION
**Pull request by n8pease**

### CODE REVIEW

File: core/modules/SConscript [hunk 1]
```diff
@@ -201,6 +201,8 @@ if not env.GetOption('clean'):
 
     env.Append(CPPPATH=[os.getcwd()])
 
+    env.Append(CCFLAGS = ' -Og ')
```
Comment (ktlim) [inline core/modules/SConscript]: You don't actually need spaces; SCons takes care of separating all CCFLAGS entries.

File: core/modules/ccontrol/ParseAdapters.h [hunk 2]
```diff
@@ -2095,7 +2095,7 @@ class AggregateWindowedFunctionAdapter :
             auto param = std::make_shared<query::ValueExpr>();
             assert_execution_condition(nullptr != _valueFactor, "ValueFactor must be populated.", _ctx);
             param->addValueFactor(_valueFactor);
-            antlr4::tree::TerminalNode * terminalNode;
+            antlr4::tree::TerminalNode * terminalNode = nullptr;
```
Comment (iagaponenko) [inline core/modules/ccontrol/ParseAdapters.h]: The `TerminalNode` times `terminalNode` is equal to `nullptr`? 😹
```
antlr4::tree::TerminalNode* terminalNode = nullptr;
```

File: core/modules/query/ValueExpr.h [hunk 3]
```diff
@@ -245,7 +245,7 @@ class ValueExpr {
 private:
     std::string _alias;
     FactorOpVector _factorOps;
-    bool _aliasIsUserDefined; /// true if the alias was defined by the user in the select statement.
+    bool _aliasIsUserDefined=false; /// true if the alias was defined by the user in the select statement.
```
Comment (iagaponenko) [inline core/modules/query/ValueExpr.h]: Also, the "LSST Style Guide" requires that "Assignment operators always have spaces around them":  https://developer.lsst.io/cpp/style.html#the-following-white-space-conventions-should-be-followed

File: core/modules/SConscript (hunk 1 above)
Comment (n8pease) [reply core/modules/SConscript]: just a typo but thx for catching.

**Pull request merged**